injectproxy/routes.go

```diff
@@ -306,18 +322,40 @@ type ctxKey int
 const keyLabel ctxKey = iota
 
 func mustLabelValue(ctx context.Context) string {
-	label, ok := ctx.Value(keyLabel).(string)
+	labels, ok := ctx.Value(keyLabel).([]string)
+	if !ok {
+		panic(fmt.Sprintf("can't find the %q value in the context", keyLabel))
+	}
+	if labels[0] == "" {
+		panic(fmt.Sprintf("empty %q value in the context", keyLabel))
+	}
+	return labels[0]
```

Comment:
I think that it's misleading and introduce inconsistency (e.g. query endpoints support enforcing multiple label values while /api/v1/silences endpoint doesn't as it only looks at the first label value). I would turn `func mustLabelValue(ctx context.Context) string` into `func mustLabelValues(ctx context.Context) []string` and update everywhere it needs to be.

In practice we need to modify

https://github.com/prometheus-community/prom-label-proxy/blob/22ccf0e32939e95ae12187c472e03afed68b7eff/injectproxy/silences.go#L46

https://github.com/prometheus-community/prom-label-proxy/blob/22ccf0e32939e95ae12187c472e03afed68b7eff/injectproxy/silences.go#L75

https://github.com/prometheus-community/prom-label-proxy/blob/22ccf0e32939e95ae12187c472e03afed68b7eff/injectproxy/silences.go#L132

https://github.com/prometheus-community/prom-label-proxy/blob/22ccf0e32939e95ae12187c472e03afed68b7eff/injectproxy/rules.go#L200

https://github.com/prometheus-community/prom-label-proxy/blob/22ccf0e32939e95ae12187c472e03afed68b7eff/injectproxy/rules.go#L226

Reply:
@simonpasquier 
I'm now trying to keep every consistent(every endpoints should support multiple label values) and there is something which I can't understand. Could you please explain to me:

Why is the first `namespace="default"` not thrown away?

https://github.com/prometheus-community/prom-label-proxy/blob/1975124010c20f5e227ea7351e0bdea798badbd5/injectproxy/routes_test.go#L288-L295

Reply:
Every endpoint can accept multiple label values now. One big change I need to mention is `/api/v2/silences` for `GET`. As I want to keep the consistency of each endpoint that it should work as `union` of all label values, I changed the procedure how the proxy filters the result. It filters the result after it received response from alertmanager, not before it sends request to alertmanager.

Please carefully review this function:
https://github.com/prometheus-community/prom-label-proxy/blob/0471c290c7f24096653d5a37e1a6a021de6ccf73/injectproxy/silences.go#L258-L311

What I want to do here is:

If we have label `namespace` and  values: `["namespace1", "namespace2", "namespace3"]`
All silences with one of the following matchers should be listed with a `GET` request:
```
# non regex:
namespace="namespace1"
namespace="namespace2"
namespace="namespace3"

# regex:
namespace=~"namespace1"
namespace=~"namespace2"
namespace=~"namespace3"
namespace=~"namespace1|namespace2"
namespace=~"namespace1|namespace3"
namespace=~"namespace2|namespace3"
namespace=~"namespace1|namespace2|namespace3"
```

And also be able to be updated/deleted with a `POST` or `DELETE` request. Because we want to have `union`.

Please tell me if there is better way to do this or I think wrong.